pkg/resourceinterpreter/default/native/aggregatestatus.go

```diff
@@ -311,12 +310,10 @@ func aggregateDaemonSetStatus(object *unstructured.Unstructured, aggregatedStatu
 
 	// The 'observedGeneration' is mainly used by GitOps tools(like 'Argo CD') to assess the health status.
 	// For more details, please refer to https://argo-cd.readthedocs.io/en/stable/operator-manual/health/.
+	newStatus.ObservedGeneration = oldStatus.ObservedGeneration
 	if observedLatestResourceTemplateGenerationCount == len(aggregatedStatusItems) {
 		newStatus.ObservedGeneration = daemonSet.Generation
-	} else {
-		newStatus.ObservedGeneration = oldStatus.ObservedGeneration
 	}
```

Comment:
Just for reduce unnecessary code of else, happy to have different opinions

Reply:
I think it's okay~